File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/rabbit/provider/PendingMessageContextHolder.java
package org.xyattic.eventual.consistency.support.core.rabbit.provider;

import com.google.common.collect.Lists;

import java.util.List;

/**
 * @author wangxing
 * @create 2020/4/1
 */
public class PendingMessageContextHolder {

    private static final ThreadLocal<List<PendingMessage>> contextHolder =
            ThreadLocal.withInitial(Lists::newArrayList);

    public static void set(List<PendingMessage> pendingMessages) {
        contextHolder.set(pendingMessages);
    }

    public static void clear() {
        contextHolder.remove();
    }

    public static List<PendingMessage> get() {
        return contextHolder.get();
    }

    public static void addAll(List<PendingMessage> pendingMessages) {
        contextHolder.get().addAll(pendingMessages);
    }

    public static void add(PendingMessage pendingMessage) {
        contextHolder.get().add(pendingMessage);
    }

}

File: eventual-consistency-support-core/src/main/java/org/xyattic/eventual/consistency/support/core/provider/PendingMessageContextHolder.kt
package org.xyattic.eventual.consistency.support.core.provider

import com.google.common.collect.Lists

/**
 * @author wangxing
 * @create 2020/4/1
 */
class PendingMessageContextHolder {
    companion object{
        private val contextHolder: ThreadLocal<MutableList<PendingMessage>> = ThreadLocal.withInitial { mutableListOf<PendingMessage>() }

        @JvmStatic
        fun set(pendingMessages: List<PendingMessage>) {
            contextHolder.set(Lists.newArrayList(pendingMessages))
        }

        @JvmStatic
        fun clear() {
            contextHolder.remove()
        }

        @JvmStatic
        fun get(): List<PendingMessage> {
            return contextHolder.get()
        }

        @JvmStatic
        fun addAll(pendingMessages: List<PendingMessage>) {
            contextHolder.get().addAll(pendingMessages)
        }

        @JvmStatic
        fun add(pendingMessage: PendingMessage) {
            contextHolder.get().add(pendingMessage)
        }

    }

}